arangod/Aql/ShardLocking.cpp

```diff
@@ -108,7 +108,7 @@ void ShardLocking::addNode(ExecutionNode const* baseNode, size_t snippetId,
       }
       auto const graphIsUsedAsSatellite = graphNode->isUsedAsSatellite();
       auto const isUsedAsSatellite = [&](auto const& col) {
-        return graphIsUsedAsSatellite || (col->isSatellite() && (pushToSingleServer || graphNode->isSmart()));
+        return graphIsUsedAsSatellite || (col->isSatellite() && (pushToSingleServer || graphNode->isSmart())) || (col->isSatellite() && graphNode->isSmart());
```

Comment:
```suggestion
        return graphIsUsedAsSatellite || (col->isSatellite() && (pushToSingleServer || graphNode->isSmart())));
```
Covered by the second OR case